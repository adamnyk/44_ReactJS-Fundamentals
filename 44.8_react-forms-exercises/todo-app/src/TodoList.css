.TodoList-todo-container {
    margin: 10px;
}