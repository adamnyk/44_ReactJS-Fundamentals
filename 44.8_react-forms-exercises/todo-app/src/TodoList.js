import { useState } from "react";
import "./TodoList.css";
import Todo from "./Todo"
import NewTodoForm from "./NewTodoForm"
import { v4 as uuid } from "uuid";


const ToDoList = () => {
    const INITIAL_TODOS_STATE = [
        // { text: 'get eggs', id: uuid() },
        // { text: 'empty compost', id: uuid() }
    ]
    const [todos, setTodos] = useState(INITIAL_TODOS_STATE)

    const removeTodo = (id) => {
        setTodos(todos => (
            todos.filter(todo => todo.id !== id)
        ))
    }

    const addTodo = (todo) => {
        if (todo.text) {
            const newTodo = {...todo, id: uuid()}
            setTodos(todos => [...todos, newTodo])
        }
    }

    const todoComponents = todos.map(todo =>  (
            <Todo
                text={todo.text}
                key={todo.id}
                id={todo.id}
                remove={removeTodo}
            />
        )
    )

    return (    
        <>
            <NewTodoForm addTodo={addTodo} />
            <div className="TodoList-todo-container">
                {todoComponents}
            </div>
        </>
    )
}

export default ToDoList