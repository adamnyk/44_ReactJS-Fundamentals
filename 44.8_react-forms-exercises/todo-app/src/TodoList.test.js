import { fireEvent, render } from '@testing-library/react';
import TodoList from './TodoList';

it('TodoList renders', () => {
    render(<TodoList />);
  });
  
it("matches snapshot", function () {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
  });

it("should add new todo", function () {
	const { queryByText, queryByTestId } = render(<TodoList />);
	const textInput = queryByTestId("text-input");
	const submitBtn = queryByText("Add");
	expect(queryByText("return library book")).not.toBeInTheDocument();
	fireEvent.change(textInput, { target: { value: "return library book" } });
	fireEvent.click(submitBtn);
	expect(queryByText("return library book")).toBeInTheDocument();

});

it("should delete a todo", function () {
	const { queryByText, queryByTestId } = render(<TodoList />);
	const textInput = queryByTestId("text-input");
	const submitBtn = queryByText("Add");
	expect(queryByText("return library book")).not.toBeInTheDocument();
	fireEvent.change(textInput, { target: { value: "return library book" } });
	fireEvent.click(submitBtn);
	expect(queryByText("return library book")).toBeInTheDocument(); 

	const deleteBtn = queryByTestId("delete-button")
	fireEvent.click(deleteBtn);
	expect(queryByText("return library book")).not.toBeInTheDocument();

});
