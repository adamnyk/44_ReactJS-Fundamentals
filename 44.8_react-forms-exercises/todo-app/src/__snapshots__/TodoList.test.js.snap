// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <div
      class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="text"
          id="text-label"
        >
          Todo
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
            data-testid="text-input"
            id="text"
            name="text"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-14lo706"
            >
              <span>
                Todo
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall css-1t6xn9b-MuiButtonBase-root-MuiButton-root"
        style="max-width: 50px; max-height: 50px; min-width: 30px; min-height: 30px; font-size: 12px;"
        tabindex="0"
        type="submit"
      >
        Add
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
  <div
    class="TodoList-todo-container"
  />
</DocumentFragment>
`;
